pkg/virt-controller/watch/vmi/vmi.go

```diff
@@ -1026,6 +1030,10 @@ func (c *Controller) sync(vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod, da
 		if validateErr := errors.Join(validateErrors...); validateErrors != nil {
 			return common.NewSyncError(fmt.Errorf("failed create validation: %v", validateErr), "FailedCreateValidation")
 		}
+		if downwardmetrics.IsDownwardMetricsConfigurationInvalid(c.clusterConfig, &vmi.Spec) {
+			c.recorder.Eventf(vmi, k8sv1.EventTypeWarning, controller.FeatureNotEnabled, downwardmetrics.DownwardMetricsNotEnabledError.Error())
+			return common.NewSyncError(fmt.Errorf("virtual machine is requesting a disabled feature: %s", "DownwardMetrics"), controller.FeatureNotEnabled)
```

Comment:
Use `downwardmetrics.DownwardMetricsNotEnabledError here too?

The use of fmt.Error with static strings looks weird.